sdk/program/src/bpf_loader_upgradeable.rs

```diff
@@ -0,0 +1,185 @@
+//! @brief An Upgradeable Solana BPF loader.
+//!
+//! The upgradeable BPF loader is responsible for deploying, upgrading, and
+//! executing BPF programs.  The upgradeable loader allows a program's authority
+//! to update the program at any time.  This ability break's the "code is law"
+//! contract the usually enforces the policy that once a program is on-chain it
+//! becomes immutable.  Because of this, care should be taken before executing
+//! upgradeable programs which still have a functioning authority.  For more
+//! information refer to `loader_upgradeable_instruction.rs`
+
+use crate::{
+    instruction::{AccountMeta, Instruction, InstructionError},
+    loader_upgradeable_instruction::UpgradeableLoaderInstruction,
+    pubkey::Pubkey,
+    system_instruction, sysvar,
+};
+use bincode::serialized_size;
+
+crate::declare_id!("BPFLoaderUpgradeab1e11111111111111111111111");
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, Copy, AbiExample)]
+pub enum UpgradableLoaderState {
+    /// Account is not initialized.
+    Uninitialized,
+    /// An executable account.
+    Executable {
+        /// Public key of the code account.
+        code_pubkey: Pubkey,
```

Comment:
nit:
```suggestion
        /// Address of the code account.
        code: Pubkey,
```

Reply:
What distinction are you trying to make?

Reply:
1) It's an address, not a pubkey.  pubkey is one kind of address
2) `_pubkey`, or `_address` suffix seems unnecessary